core/src/main/java/me/aurium/beetle/api/task/Comparisons.java

```diff
@@ -0,0 +1,36 @@
+package me.aurium.beetle.api.task;
+
+import me.aurium.beetle.api.task.futures.TaskFuture;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * i give up
```

Comment:
Remember, this is supposed to be fun, and this is your framework. If you don't want to make a certain improvement you don't have to. https://bill.burkecentral.com/2021/03/31/anybody-else-hate-pr-reviews/

In all seriousness, it probably isn't necessary for you to check Comparisons.CONSTANT. That's just a generics trick I used for a very small yet satisfying performance gain.

Reply:
I think this pr is ready to merge haha but i also want to make sure you've seen and verified every aspect of the code before beginning to use it in things like solar, qteam, and opencode (all production environments)